tests/servers/test_basehttp.py

```diff
@@ -189,3 +190,27 @@ def test_broken_pipe_errors(self):
                         self.assertEqual(cm.records[0].getMessage(), msg)
                 finally:
                     server.server_close()
+
+
+@override_settings(ROOT_URLCONF="servers.urls")
+class LiveServerViews(LiveServerTestCase):
+    available_apps = []
+
+    def test_head_content_length_removed(self):
+        conn = HTTPConnection(
+            LiveServerViews.server_thread.host, LiveServerViews.server_thread.port
+        )
```

Comment:
Do we need to use `LiveServer` in tests? :thinking:

Reply:
`ServerHandler` where the change was made is used by `runserver` (`WSGIServer`) and `LiveServer`. I couldn't find any tests that directly use `WSGIServer` and make HTTP calls the were similar to what I wanted but I did find the `LiveServer` calls. That's why I stuck it originally where I had it. Those `LiveServer` tests are basically how I wanted to test it.

So we have to use `LiveServer` or `WSGIServer` or maybe some kind of unit test on `ServerHandler`. I went the with the path that had tests already that I could base mine off of. I could probably figure out testing `ServerHandler` or `WSGIServer` but it was a longer task t figure out and `LiveServer` tested the code just fine.

Let me know your recommendations.

Reply:
`test_head_content_length_removed` scenario is already covered by `servers.test_basehttp.WSGIRequestHandlerTestCase.test_no_body_returned_for_head_requests`, so it should be enough to add a new `WSGIRequestHandler` test (see https://github.com/django/django/pull/17643#discussion_r1437997140).